src/infrastructure/figma/figma-service.ts

```diff
@@ -159,6 +164,49 @@ export class FigmaService {
 			accessToken,
 		});
 	};
+
+	createFileUpdateWebhook = async (
+		teamId: string,
+		atlassianUserId: string,
+		connectInstallationSecret: string,
+	): Promise<{ webhookId: string; teamId: string }> => {
+		const { accessToken } =
+			await this.getValidCredentialsOrThrow(atlassianUserId);
+
+		const passcode = this.generateWebhookPasscode({
+			atlassianUserId,
+			figmaTeamId: teamId,
+			connectInstallationSecret,
+		});
+
+		const request: CreateWebhookRequest = {
+			event_type: 'FILE_UPDATE',
+			team_id: teamId,
+			endpoint: `${getConfig().app.baseUrl}/figma/webhook`,
+			passcode,
+			description: 'Figma for Jira Cloud',
+		};
+
+		const result = await figmaClient.createWebhook(request, accessToken);
+		return { webhookId: result.id, teamId: result.team_id };
+	};
+
+	validateWebhookPasscode = (passcode: string, input: WebhookPasscodeInput) => {
+		return passcode === this.generateWebhookPasscode(input);
+	};
+
+	/**
+	 * @internal
+	 * Visible for testing only.
+	 */
+	generateWebhookPasscode = ({
+		atlassianUserId,
+		figmaTeamId,
+		connectInstallationSecret,
+	}: WebhookPasscodeInput): string => {
+		const input = atlassianUserId + figmaTeamId + connectInstallationSecret;
```

Comment:
**suggestion (non-blocking):** It might be better to keep a Connect App installation shared secret exclusively between the Connect App and Jira.

Here is what we could do instead:

1. Generate a UUID and store it in a new `FigmaTeam.webhookPasscode` column. This solution also requires querying a database on each webhook request.
2. Sign `FigmaTeam.teamId` (which is passed to `POST/v2/webhooks`) with a public key stored in the environment variables. This solution does not require querying a database on each webhook request.

Let's discuss this when you are available, Justin. Anyway, it is not a blocker for this PR -- we can address this in a separate ticket (if needed).

Reply:
We've decided to go with option 1 here, to generate a UUID for the passcode and store it in the `FigmaTeam` table.